internals/schema_migration.py

```diff
@@ -22,14 +23,23 @@ class MigrateCommentsToActivities(FlaskHandler):
   def get_template_data(self):
     """Writes an Activity entity for each unmigrated Comment entity."""
     self.require_cron_header()
+
+    logging.info(self._remove_bad_id_activities())
+
     q = Comment.query()
     comments = q.fetch()
     migration_count = 0
     for comment in comments:
-      if comment.migrated:
+      # Check if an Activity with the same ID has already been created.
```

Comment:
Overwriting would have no net effect in that case, so it should be OK.

If you want to avoid overwriting, you could make this loop more scalable by first fetching all Activities with keys_only=True and then make a python set with those IDs.  It would be one round-trip to ndb instead of N.